pkg/cmd/cli/cmd/push/binary.go

```diff
@@ -0,0 +1,368 @@
+package push
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapiv1 "k8s.io/kubernetes/pkg/api/v1"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+	"github.com/openshift/origin/pkg/image/apis/image/docker10"
+	"github.com/openshift/origin/pkg/image/apis/image/v1"
+	clientset "github.com/openshift/origin/pkg/image/generated/clientset/typed/image/v1"
+)
+
+var (
+	pushBinaryLong = templates.LongDesc(`
+		Create a new image by pushing an archive
+
+		This command assists users in creating new images that can be used as part of builds
+		or as deployed applications. It accepts a zip or tar.gz file that will become a new 
+		image in an image stream. You may also specify image metadata like the entrypoint, 
+		environment variables, or labels to create a runnable image.
```

Comment:
there's an implicit assumption here that the archive is going to be extracted when it is used to create the image layer.  might want to make that explicit.  (you get an image containing the contents of the zip, not an image containing the zip)